File: benchmarks/pandas/bench_stack.py
"""Benchmark: stack — df.stack() converts a wide DataFrame to long Series on a 1000x10 DataFrame"""
import json, time
import numpy as np
import pandas as pd

ROWS = 1_000
COLS = 10
WARMUP = 5
ITERATIONS = 50

data = {f"col{c}": np.arange(ROWS) * COLS + c for c in range(COLS)}
df = pd.DataFrame(data)

for _ in range(WARMUP):
    df.stack()

start = time.perf_counter()
for _ in range(ITERATIONS):
    df.stack()
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "stack",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_stack.ts
/**
 * Benchmark: stack — stack(df) converts a wide DataFrame to a long Series on a 1000x10 DataFrame
 */
import { DataFrame, stack } from "../../src/index.js";

const ROWS = 1_000;
const COLS = 10;
const WARMUP = 5;
const ITERATIONS = 50;

const colData: Record<string, number[]> = {};
for (let c = 0; c < COLS; c++) {
  colData[`col${c}`] = Array.from({ length: ROWS }, (_, i) => i * COLS + c);
}
const df = DataFrame.fromColumns(colData);

for (let i = 0; i < WARMUP; i++) {
  stack(df);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  stack(df);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "stack",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
